task/sql-tasks.js

```diff
@@ -280,7 +460,24 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT DISTINCT
+        c.CompanyName as "CompanyName",
+        p.ProductName as "ProductName",
+        od.UnitPrice as "PricePerItem"
+    FROM Products p 
+    JOIN OrderDetails od ON p.ProductID = od.ProductID 
+    JOIN Orders o ON od.OrderID = o.OrderID 
+    JOIN Customers c ON o.CustomerID = c.CustomerID
+    WHERE od.UnitPrice = (SELECT 
+                            MAX(od2.UnitPrice)
+                        FROM Customers c1
+                        JOIN Orders o2 ON o2.CustomerID = c1.CustomerID 
+                        JOIN OrderDetails od2 ON o2.OrderID = od2.OrderID 
+                        WHERE c1.CompanyName = c.CompanyName) 

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.